### PR TITLE
HotFix: fix compare function for DateTime type

### PR DESCRIPTION
**Pull request by Naupio**
elixir Enum.sort_by default sorter `asc` and `desc`  which is a convenience for [`&<=/2`](https://hexdocs.pm/elixir/1.13/Kernel.html#%3C=/2) and [`&>=/2`](https://hexdocs.pm/elixir/1.13/Kernel.html#%3E=/2) respectively. 

in erlang world ,all of `term` is comparable

In this pr scenario, we need compare `updated_at` field by sorting return data and distinct it.

but the `DateTime` compare with default sorter not return correct result
it cause some error like this
```
iex(1)> ~U[2022-07-25 07:53:57.788000Z] > ~U[2022-09-07 10:34:14.021000Z]
true

iex(2)> DateTime.utc_now
~U[2022-09-17 04:00:49.814911Z]
iex(3)> ~U[2022-09-17 04:00:49.814911Z] > ~U[2022-09-07 10:34:14.021000Z]
true
```

so we need add a custom sorter to replace the default sorter `asc/desc` by using `DateTime` compare function
```
iex(4)> DateTime.compare(~U[2022-07-25 07:53:57.788000Z], ~U[2022-09-07 10:34:14.021000Z])
:lt
iex(5)> DateTime.compare(~U[2022-09-17 04:00:49.814911Z], ~U[2022-09-07 10:34:14.021000Z])
:gt                      
iex(6)> 
```

### CODE REVIEW

**Comment by Keith-CY** [comment]
Please add a brief description of this change, e.g. why use a custom comparison function instead of the default one.

**Comment by Naupio** [comment]
> Please add a brief description of this change, e.g. why use a custom comparison function instead of the default one.

added

> elixir Enum.sort_by default sorter `asc` and `desc` which is a convenience for [`&<=/2`](%60%3C=/2%60) and [`&>=/2`](%60%3E=/2%60) respectively.
> 
> in erlang world ,all of `term` is comparable
> 
> In this pr scenario, we need compare `updated_at` field by sorting return data and distinct it. it cause some error like this
> 
> ```
> iex(1)> ~U[2022-07-25 07:53:57.788000Z] > ~U[2022-09-07 10:34:14.021000Z]
> true
> 
> iex(2)> DateTime.utc_now
> ~U[2022-09-17 04:00:49.814911Z]
> iex(3)> ~U[2022-09-17 04:00:49.814911Z] > ~U[2022-09-07 10:34:14.021000Z]
> true
> ```
> 
> so we need add a custom sorter to replace the default sorter `asc/desc`
> 
> ```
> iex(4)> DateTime.compare(~U[2022-07-25 07:53:57.788000Z], ~U[2022-09-07 10:34:14.021000Z])
> :lt
> iex(5)> DateTime.compare(~U[2022-09-17 04:00:49.814911Z], ~U[2022-09-07 10:34:14.021000Z])
> :gt                      
> iex(6)> 
> ```

**Comment by Naupio** [comment]
> Please add a brief description of this change, e.g. why use a custom comparison function instead of the default one.

added



> elixir Enum.sort_by default sorter `asc` and `desc` which is a convenience for [`&<=/2`](https://hexdocs.pm/elixir/1.13/Kernel.html#%3C=/2) and [`&>=/2`](https://hexdocs.pm/elixir/1.13/Kernel.html#%3E=/2) respectively.
> 
> in erlang world ,all of `term` is comparable
> 
> In this pr scenario, we need compare `updated_at` field by sorting return data and distinct it.
> 
> but the `DateTime` compare with default sorter not return correct result it cause some error like this
> 
> ```
> iex(1)> ~U[2022-07-25 07:53:57.788000Z] > ~U[2022-09-07 10:34:14.021000Z]
> true
> 
> iex(2)> DateTime.utc_now
> ~U[2022-09-17 04:00:49.814911Z]
> iex(3)> ~U[2022-09-17 04:00:49.814911Z] > ~U[2022-09-07 10:34:14.021000Z]
> true
> ```
> 
> so we need add a custom sorter to replace the default sorter `asc/desc` by using `DateTime` compare function
> 
> ```
> iex(4)> DateTime.compare(~U[2022-07-25 07:53:57.788000Z], ~U[2022-09-07 10:34:14.021000Z])
> :lt
> iex(5)> DateTime.compare(~U[2022-09-17 04:00:49.814911Z], ~U[2022-09-07 10:34:14.021000Z])
> :gt                      
> iex(6)> 
> ```

**Review by Keith-CY — APPROVED** [review]

File: test/graphql/account_udt_test.exs [hunk 1]
```diff
@@ -29,15 +29,19 @@ defmodule GodwokenExplorer.Graphql.AccountUDTTest do
 
     user = insert!(:user)
 
-    base_t = DateTime.utc_now()
-    inc_base_t = DateTime.add(base_t, 10, :second)
-    fetch_base_t = DateTime.add(base_t, -10, :second)
+    # base_t = DateTime.utc_now()
```
Comment (zmcNotafraid) [inline test/graphql/account_udt_test.exs:32]: 注释删掉

**Review by zmcNotafraid — APPROVED** [review]

**Pull request merged**